pints/_abc/__init__.py

```diff
@@ -0,0 +1,366 @@
+#
+# Sub-module containing ABC inference routines
+#
+# This file is part of PINTS (https://github.com/pints-team/pints/) which is
+# released under the BSD 3-clause license. See accompanying LICENSE.md for
+# copyright notice and full license details.
+#
+from __future__ import absolute_import, division
+from __future__ import print_function, unicode_literals
+
+import numpy as np
+
+import pints
+
+
+class ABCSampler(pints.Loggable, pints.TunableMethod):
+    """
+    Abstract base class for Approximate Bayesian Computation (ABC) methods.
+
+    All ABC samplers implement the :class:`pints.Loggable` and
+    :class:`pints.TunableMethod` interfaces.
+    """
+    def name(self):
+        """
+        Returns this method's full name.
+        """
+        raise NotImplementedError
+
+    def ask(self):
+        """
+        Returns a parameter vector sampled from a prior
+        """
+        raise NotImplementedError
+
+    def tell(self, fx):
+        """
+        Performs an iteration of the ABC-rejection algorithm, using the
+        parameters specified by ask.
+
+        Returns the accepted parameter values, or ``None`` to indicate
+        that no parameters were accepted (tell allows for multiple evaluations
+        per iteration).
+
+        """
+        raise NotImplementedError
+
+
+class ABCController(object):
+    """
+    Samples from a :class:`pints.LogPrior`.
+
+    Properties related to the number of iterations, parallelisation,
```

Comment:
Before going into this, give a _very_ brief idea of what's going to happen. E.g. what is the "target number of samples", and what does the threshold do?

Reply:
After that, but still before this paragraph, please add a code example where the controller is created

Reply:
Also seems threshold is _not_ set here?